Userland/Libraries/LibGUI/Application.cpp

```diff
@@ -388,7 +388,7 @@ void Application::set_most_recently_open_file(ByteString new_path)
 
     new_recent_files_list.prepend(new_path);
 
```

Comment:
Actually, I think a *marginally* better fix might be to put `new_recent_files_list.resize(max_recently_open_files());` here. This would ensure that the old list is completely overwritten with any duplicates removed.

Bearing in mind that either fix only comes into play if you try to insert an empty entry, or manually edit the config file to insert duplicates.

Reply:
I believe I made the necessary changes, though I think I misunderstand something here because with resizing it seems to me that it would add empty entries if new list < max_recently_open_files() (though I do see that first loop will return empty file paths in the list if there isn't any recent opened files, so I guess it's fine?). Thanks for your help!

Reply:
The recent files list treats a blank entry as an empty slot. We are deliberately adding empty entries in the case that there are duplicates, so that the slots they took in the list are freed.

Reply:
Makes perfect sense to me, thanks!